src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java

```diff
@@ -664,4 +674,49 @@ private String postRequest(String path, NameValuePair[] params) throws IOExcepti
         }
         return branches;
     }
+
+    public Iterable<SCMFile> getDirectoryContent(final BitbucketSCMFile parent) throws IOException, InterruptedException {
+        StringBuilder url = new StringBuilder(V2_API_BASE_URL);
+        url.append(owner);
+        url.append('/');
+        url.append(repositoryName);
+        url.append("/src/");
+        url.append(Util.rawEncode(parent.getRef()));
+        url.append('/');
+
+        for (String segment : StringUtils.split(parent.getPath(), "/")) {
+            url.append(Util.rawEncode(segment));
+            url.append('/');
+        }
+        List<SCMFile> result = new ArrayList<>();
+        String response = getRequest(url.toString());
+        BitbucketCloudPage<BitbucketRepositorySource> page = JsonParser.mapper.readValue(response,
+                new TypeReference<BitbucketCloudPage<BitbucketRepositorySource>>(){});
+
+        for(BitbucketRepositorySource source:page.getValues()){
+            result.add(source.toBitbucketScmFile(parent));
+        }
+
+        while (!page.isLastPage()){
+            response = getRequest(page.getNext());
+            page = JsonParser.mapper.readValue(response,
+                    new TypeReference<BitbucketCloudPage<Map>>(){});
+            for(BitbucketRepositorySource source:page.getValues()){
+                result.add(source.toBitbucketScmFile(parent));
+            }
+        }
+        return result;
+    }
+
+    public InputStream getFileContent(BitbucketSCMFile file) throws IOException, InterruptedException {
+        String url = V2_API_BASE_URL + owner +
+                '/' +
+                repositoryName +
+                "/src/" +
+                Util.rawEncode(file.getRef()) +
+                '/' +
+                file.getPath();
```

Comment:
Does this path need splitting on `/`, rawEncode and join with `/` again?

Reply:
Good catch. fixing...